lib/nested_record/lookup_const.rb

```diff
@@ -2,17 +2,34 @@
 
 module NestedRecord
   # Derived from Rails source: https://github.com/rails/rails/blob/98a57aa5f610bc66af31af409c72173cdeeb3c9e/activerecord/lib/active_record/inheritance.rb#L181-L207
+  def self.constantize(type_name)
+    # TODO: Remove this when we drop support for Rails < 7.0
+    if ActiveSupport::Dependencies.respond_to?(:constantize)
+      ActiveSupport::Dependencies.constantize(type_name)
+    else
+      type_name.constantize
+    end
+  end
+
+  def self.safe_constantize(type_name)
+    # TODO: Remove this when we drop support for Rails < 7.0
+    if ActiveSupport::Dependencies.respond_to?(:safe_constantize)
+      ActiveSupport::Dependencies.safe_constantize(type_name)
+    else
+      type_name.safe_constantize
+    end
+  end
```

Comment:
```suggestion
  # TODO: Remove this when we drop support for Rails < 7.0
  if ActiveSupport::VERSION::MAJOR < 7
    def self.constantize(type_name)
      ActiveSupport::Dependencies.constantize(type_name)
    end
    
    def self.safe_constantize(type_name)
      ActiveSupport::Dependencies.safe_constantize(type_name)
    end
  else
    def self.constantize(type_name)
      type_name.constantize
    end
    
    def self.safe_constantize(type_name)
      type_name.safe_constantize
    end
  end
```

Reply:
@mkhairi Yes it works but `ActiveSupport::Dependencies.constantize` on older Rails leverages the `ActiveSupport::Dependencies::ClassCache`. We should do it too.

Reply:
@mkhairi Sure, please squash the commits